Automattic/woocommerce-payments
# Highlight the active dispute task if disputes due within 72 hours.

Fixes #6701

#### Changes proposed in this Pull Request



This PR changes the criteria used to determine if the Active Disputes task on the **Payments → Overview** screen is coloured yellow or red.

* Any disputes due within 72h = 🔴 Red
* Otherwise = 🟡 Yellow

I've also fixed a minor bug in the dispute task `isDueWithin()` logic – it will now correctly return `false` if it is not within the desired number of days/hours.

> Previously, it would return `true` if asked for disputes within 3 days and a dispute was due between 3-4 days / 72-96 hours.
> In this PR, it will return `false` in this case, since a dispute due in 3-4 days is not due within 3 days.

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/c540b9d6-b839-4de6-8d9b-bf45e578a610)


#### Testing instructions






* To create a disputed transaction, purchase a product with the Stripe test card `4000000000000259`.
* View the Payments → Overview screen with the following configuration of disputes and confirm it matches the requirements above:
	* 1 or more active disputes due within 7 days but more than 72 hours 
	* 1 or more active disputes due within 72 hours 

**Mocking dispute due_by when testing**

To mock dispute `due_by` dates testing purposes, add the following logic and adjust the 'days' value: https://github.com/Automattic/woocommerce-payments/blob/56dd736e4b2d335ffcdd4e4a6b676d7df0861c78/client/overview/index.js#L75
```js
	activeDisputes: activeDisputes.map( ( dispute ) => {
		const hours = 73;
		dispute.due_by = new Date(
			Date.now() + hours * 60 * 60 * 1000
		).toISOString();
		return dispute;
	} );
```

-------------------


**Post merge**
